### DISCUSSION TITLE
ERROR: Memory allocation on device failed

### DISCUSSION THREAD

**Discussion by XueZhengHe03** [ExtremeFLOW/neko#1089] [Q&A] [answered]
when I mpirun tgv_RE1600 on NVIDIA GPU , it stiil have the warning, The neko version is 0.6.1 . 
*** ERROR: Memory allocation on device failed ***
Warning: ieee_underflow is signaling
Warning: ieee_inexact is signaling
ERROR STOP

**Comment by njansson** [comment]
That would indicate that your device can't fit the entire problem size. Either use a device with more memory, or run across multiple devices, by using more mpi ranks, and map each rank to an own devices via a wrapper script. See for example
https://github.com/ExtremeFLOW/flettner_rotor/blob/main/run.sh

**Reply by XueZhengHe03** [reply]
I have used the way as  https://github.com/ExtremeFLOW/flettner_rotor/blob/main/run.sh, and I also using more mpi ranks ,but it still cant  
work ,maybe I need  how much memory?

**Reply by njansson** [reply, marked as answer]
Depends on which gpu you have. One or two A100 should be enough for that case. Other problem sources could be that mpi ranks are mapped to the same card
